presto-main/src/test/java/com/facebook/presto/cost/TestScalarStatsCalculator.java

```diff
@@ -73,6 +89,334 @@ public void setUp()
         translator = new TestingRowExpressionTranslator(MetadataManager.createTestMetadataManager());
     }
 
+    public static final class CustomFunctions
+    {
+        private CustomFunctions() {}
+
+        @ScalarFunction(value = "custom_add", calledOnNullInput = false)
+        @ScalarFunctionConstantStats(avgRowSize = 8.0)
+        @SqlType(StandardTypes.BIGINT)
+        public static long customAdd(
+                @ScalarPropagateSourceStats(
+                        propagateAllStats = false,
+                        nullFraction = StatsPropagationBehavior.SUM_ARGUMENTS,
+                        distinctValuesCount = StatsPropagationBehavior.SUM_ARGUMENTS,
+                        minValue = StatsPropagationBehavior.SUM_ARGUMENTS,
+                        maxValue = StatsPropagationBehavior.SUM_ARGUMENTS) @SqlType(StandardTypes.BIGINT) long x,
+                @SqlType(StandardTypes.BIGINT) long y)
+        {
+            return x + y;
+        }
+
+        @ScalarFunction(value = "custom_is_null", calledOnNullInput = true)
+        @LiteralParameters("x")
+        @SqlType(StandardTypes.BOOLEAN)
+        @ScalarFunctionConstantStats(distinctValuesCount = 2.0, nullFraction = 0.0)
```

Comment:
Can we set `distinctValuesCount` to some value other than `2.0` which is also used in the `custom_is_null` BIGINT variant. GIves us confidence that we are picking the value from the actual annotated header